minidump-processor/src/op_analysis.rs

```diff
@@ -154,222 +234,341 @@ mod amd64 {
+                | CommonOpcode::IRETQ
+                | CommonOpcode::POP
+                | CommonOpcode::RETF
+                | CommonOpcode::RETURN => {
+                    if let Ok(rsp) = context.get_regspec(RegSpec::rsp()) {
+                        push_implicit_access(rsp, MemoryAccessType::Read);
+                    }
+                }
+                _ => (),
+            }
+            Ok(())
+        }
+
+        fn add_uncommon_opcode_accesses(
+            accesses: &mut Vec<MemoryAccess>,
+            instruction: Instruction,
+            context: &MinidumpContext,
+        ) -> Result<(), OpAnalysisError> {
+            // Shortcut -- If the instruction doesn't access memory, just return
+            let mem_size = match instruction.mem_size() {
+                Some(access) => access.bytes_size(),
+                None => return Ok(()),
+            };
+
+            for idx in 0..instruction.operand_count() {
+                Self::add_uncommon_opcode_explicit_access(
+                    accesses,
+                    instruction.operand(idx),
+                    mem_size,
+                    context,
+                )?;
+            }
+
+            Ok(())
+        }
+        fn add_uncommon_opcode_explicit_access(
+            accesses: &mut Vec<MemoryAccess>,
+            operand: Operand,
+            mem_size: Option<u8>,
+            context: &MinidumpContext,
+        ) -> Result<(), OpAnalysisError> {
+            if !operand.is_memory() {
+                return Ok(());
+            }
+
+            if let Some(op_info) = MemoryOperandInfo::try_from_operand(operand) {
+                let address_info = MemoryAddressInfo::from_memory_operand_info(op_info, context)?;
+                accesses.push(MemoryAccess {
+                    address_info,
+                    size: mem_size,
+                    access_type: MemoryAccessType::UncommonInstructionAccess,
+                });
+            }
+
+            Ok(())
+        }
+    }
+
+    impl InstructionPointerUpdate {
+        fn from_instruction(
+            instruction: Instruction,
+            context: &MinidumpContext,
+            memory_list: Option<&minidump::UnifiedMemoryList>,
+            stack_memory: Option<minidump::UnifiedMemory>,
+        ) -> Result<Option<Self>, OpAnalysisError> {
+            let rip_update = |address| {
+                Some(InstructionPointerUpdate {
+                    address_info: MemoryAddressInfo {
+                        address,
+                        is_likely_null_pointer_dereference: address == 0,
+                        is_likely_guard_page: false,
+                    },
+                })
+            };
+
+            match instruction.opcode() {
                 Opcode::CALL | Opcode::CALLF | Opcode::JMP | Opcode::JMPF | Opcode::JMPE => {
-                    if decoded_instruction.operand_count() != 1 {
+                    if instruction.operand_count() != 1 {
```

Comment:
This actually seems worth using `assert!()` for -- If this is false, there is a bug in Yaxpeax that needs fixing and we want it to fail loudly.